types.go

```diff
@@ -4,7 +4,15 @@ import "context"
 
 // SortType defines the interface required by the extsort library to be able to sort the items
 type SortType interface {
-	ToBytes() []byte // ToBytes used for marshaling with gob
+	ToBytes(buf []byte) []byte
+}
+
+type Sizer interface {
+	Size() int
+}
+
+type Releaser interface {
```

Comment:
Maybe should add a little doc here and on Sizer too.

Reply:
Yes. I don't understand why they were added.